sed/loader/flash/buffer_handler.py

```diff
@@ -149,16 +146,17 @@ def _save_buffer_file(self, paths: dict[str, Path]) -> None:
 
         # Reset the index of the DataFrame and save both the electron and timed dataframes
         # electron resolved dataframe
-        electron_channels = get_channels(self._config["channels"], "per_electron")
-        dtypes = get_dtypes(self._config["channels"], "all")
+        electron_channels = get_channels(self._config, "per_electron")
+        dtypes = get_dtypes(self._config["channels"], df.columns.values)
```

Comment:
It would be consistent to also change the first argument of get_dtypes to accept the whole config.